vetomint/src/progress.rs

```diff
@@ -1,34 +1,36 @@
 use super::*;
 
 /// TODO: we have to implement the following missing logics
-/// - on-proposal
+/// - on-proposal -> search TBC
 /// - on-4f-favor-prevote-propose-step
 /// - on-4f-favor-prevote-prevote-step
-/// - on-4f-nil-prevote
+/// - on-4f-nil-prevote -> to be checked
 /// - on-5f-precommit
 /// - on-4f-favor-precommit
-/// - OnTimeoutPrecommit
+/// - OnTimeoutPrecommit -> to be checked
 pub(super) fn progress(
     height_info: &HeightInfo,
     state: &mut ConsensusState,
     event: ConsensusEvent,
 ) -> Option<Vec<ConsensusResponse>> {
     let result = if state.waiting_for_proposal_creation {
-        if let ConsensusEvent::BlockProposalCreated {
-            proposal, round, ..
-        } = event
-        {
-            if state.round != round {
+        match event {
```

Comment:
why don't you use `if let` syntax?

Reply:
To make a same form with else statement of result, I adopted match, but I am not sure is there any advantage using match.
If `if let` syntax has other advantage(looks better or performance), I would gladly change it!

Reply:
ok